### PR TITLE
Add GetServiceWithName and GetServicesPortDetails

### PR DESCRIPTION
**Pull request by stuggi**
Adds functions to set a Service with name in namespace and one to get a ServicePort with its name from the service.

### CODE REVIEW

File: modules/common/service/service.go [hunk 1]
```diff
@@ -176,3 +177,40 @@ func GetServicesListWithLabel(
...
+		return nil, err
+	}
+
+	return service, nil
+}
+
+// GetServicesPortDetails - Return ServicePort with name from a Service
+func GetServicesPortDetails(
+	ctx context.Context,
+	h *helper.Helper,
```
Comment (abays) [inline modules/common/service/service.go:204]: Doesn't look like we need the context or the helper here

Comment (stuggi) [reply modules/common/service/service.go:204]: right, will remove it

File: modules/common/service/service.go [hunk 2]
```diff
@@ -176,3 +177,38 @@ func GetServicesListWithLabel(
...
+func GetServiceWithName(
+	ctx context.Context,
+	h *helper.Helper,
+	name string,
+	namespace string,
+) (*corev1.Service, error) {
+
+	service := &corev1.Service{}
+
+	err := h.GetClient().Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, service)
```
Comment (abays) [inline modules/common/service/service.go:191]: Just one question.  You mentioned in the other PR (https://github.com/openstack-k8s-operators/lib-common/pull/158) that you wanted to use KClient to not be restricted to cache-only pods in watched namespaces.  Do we have any interest in doing that here too?

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**